include/cinatra/coro_http_client.hpp

```diff
@@ -304,17 +322,39 @@ class coro_http_client {
       assert(false);
 #endif
     }
+    // construct proxy request uri
+    if (!proxy_host_.empty() && !proxy_port_.empty()) {
+      if (!proxy_request_uri_.empty())
+        proxy_request_uri_.clear();
+      if (u.get_port() == "http") {
+        proxy_request_uri_ += "http://" + u.get_host() + ":";
+        proxy_request_uri_ += "80";
+      }
+      else if (u.get_port() == "https") {
+        proxy_request_uri_ += "https://" + u.get_host() + ":";
+        proxy_request_uri_ += "443";
+      }
+      else {
+        // all be http
+        proxy_request_uri_ += " http://" + u.get_host() + ":";
+        proxy_request_uri_ += u.get_port();
+      }
+      proxy_request_uri_ += u.get_path();
+      u.path = std::string_view(proxy_request_uri_);
+    }
```

Comment:
could you extract such code into a method?

Reply:
finished it.now handle_uri only add one statement.